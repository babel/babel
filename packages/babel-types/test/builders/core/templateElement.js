import * as t from "../../../lib/index.js";

describe("builders", function () {
  describe("core", function () {
    describe("templateElement", function () {
      it("should validate", function () {
        expect(
          t.templateElement({ raw: "foo", cooked: "foo" }),
        ).toMatchSnapshot();

        expect(t.templateElement({ raw: "foo" })).toMatchSnapshot();

        expect(() =>
          t.templateElement({ raw: 1 }),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateElement({ raw: "foo", cooked: 1 }),
        ).toThrowErrorMatchingSnapshot();

        expect(() => t.templateElement("foo")).toThrowErrorMatchingSnapshot();

        expect(t.templateElement({ raw: "\\u" })).toMatchSnapshot();

        expect(t.templateElement({ raw: "\\x42" })).toMatchSnapshot();

        expect(
          t.templateElement({ raw: "\\x42", cooked: "123" }),
        ).toMatchSnapshot();

        expect(() =>
          t.templateElement({ raw: "`" }),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateElement({ raw: "${" }),
        ).toThrowErrorMatchingSnapshot();
      });
    });
    describe("templateLiteral", function () {
      it("should validate", function () {
        const foo = t.templateElement({ raw: "foo" });
        const bar = t.templateElement({ raw: "bar" });

        const baz = t.stringLiteral("baz");
        const qux = t.stringLiteral("qux");

        expect(t.templateLiteral([foo], [])).toMatchSnapshot();

        expect(t.templateLiteral([foo, bar], [baz])).toMatchSnapshot();

        expect(() =>
          t.templateLiteral([foo, bar], [baz, qux]),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateLiteral([foo, bar], []),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateLiteral({}, [baz]),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateLiteral([foo, bar]),
        ).toThrowErrorMatchingSnapshot();
      });
    });
  });
});
