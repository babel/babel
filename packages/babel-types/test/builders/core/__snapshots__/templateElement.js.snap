// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`builders core templateElement should validate 1`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "cooked": "foo",
    "raw": "foo",
  },
}
`;

exports[`builders core templateElement should validate 2`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "cooked": "foo",
    "raw": "foo",
  },
}
`;

exports[`builders core templateElement should validate 3`] = `
"Property value of TemplateElement expected to have the following:
Property raw expected type of string but got number"
`;

exports[`builders core templateElement should validate 4`] = `
"Property value of TemplateElement expected to have the following:
Property cooked expected type of string but got number"
`;

exports[`builders core templateElement should validate 5`] = `
"Property value of TemplateElement expected to have the following:
Property raw expected type of string but got undefined"
`;

exports[`builders core templateElement should validate 6`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "cooked": null,
    "raw": "\\\\u",
  },
}
`;

exports[`builders core templateElement should validate 7`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "cooked": "B",
    "raw": "\\\\x42",
  },
}
`;

exports[`builders core templateElement should validate 8`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "cooked": "B",
    "raw": "\\\\x42",
  },
}
`;

exports[`builders core templateElement should validate 9`] = `"Invalid raw"`;

exports[`builders core templateElement should validate 10`] = `"Invalid raw"`;

exports[`builders core templateLiteral should validate 1`] = `
Object {
  "expressions": Array [],
  "quasis": Array [
    Object {
      "tail": false,
      "type": "TemplateElement",
      "value": Object {
        "cooked": "foo",
        "raw": "foo",
      },
    },
  ],
  "type": "TemplateLiteral",
}
`;

exports[`builders core templateLiteral should validate 2`] = `
Object {
  "expressions": Array [
    Object {
      "type": "StringLiteral",
      "value": "baz",
    },
  ],
  "quasis": Array [
    Object {
      "tail": false,
      "type": "TemplateElement",
      "value": Object {
        "cooked": "foo",
        "raw": "foo",
      },
    },
    Object {
      "tail": false,
      "type": "TemplateElement",
      "value": Object {
        "cooked": "bar",
        "raw": "bar",
      },
    },
  ],
  "type": "TemplateLiteral",
}
`;

exports[`builders core templateLiteral should validate 3`] = `
"Number of TemplateLiteral quasis should be exactly one more than the number of expressions.
Expected 3 quasis but got 2"
`;

exports[`builders core templateLiteral should validate 4`] = `
"Number of TemplateLiteral quasis should be exactly one more than the number of expressions.
Expected 1 quasis but got 2"
`;

exports[`builders core templateLiteral should validate 5`] = `"Property quasis expected type of array but got object"`;

exports[`builders core templateLiteral should validate 6`] = `"Property expressions expected type of array but got undefined"`;
